### PR TITLE
Add initialisers to vendor identity tokens 

### LINKED ISSUE #131: Can't create `GoogleIdentityToken` without a Decoder
**Is your feature request related to a problem? Please describe.**
Often times we provide properties of `GoogleIdentityToken` manually, as we get them through Google Docs etc, not from a JSON body. But because it extends `Decodable`, the automatic initializer isn't available, so the only way to create it (AFAIK) is to create a custom `Decodable` object, which is awkward and bad in performance. We can't even extend it with our own initializer because it contains `let` properties.

**Describe the solution you'd like**
An initializer could be added manually, if no less-boilerplate solution is found.

**Describe alternatives you've considered**
`GoogleIdentityTokenFactory` object (with the same properties as `GoogleIdentityToken`) that doesn't conform to `Decodable` (thus has the automatic memberwise initializer) and has a method that returns `GoogleIdentityToken` (or `GoogleIdentityToken` could have an initializer taking `GoogleIdentityTokenFactory`, for discoverability). I think this is overengineering it though, and doesn't provide less boilerplate.

**Additional context**
I may be missing something, because this looks like a very commonly used thing.

### PR DESCRIPTION
**Pull request by ptoffy**
Fixes #131. Also adds tests to the vendor token payloads to get some coverage up

### CODE REVIEW

File: Sources/JWTKit/Claims/TenantIDClaim.swift [hunk 1]
```diff
@@ -7,12 +7,20 @@
...
     public init(value: String?) {
         self.value = value
     }
+
+    public init(stringLiteral value: String) {
+        self.value = value
+    }
+
+    public init(nilLiteral: ()) {
+        self.value = nil
```
Comment (gwynne) [inline Sources/JWTKit/Claims/TenantIDClaim.swift:24]: Delegate to the existing initializer.
```suggestion
        self.init(value: nil)
```

File: Sources/JWTKit/Claims/TenantIDClaim.swift [hunk 2]
```diff
@@ -7,12 +7,20 @@
 /// logging or auditing the tenant context of the authenticated user. The value of "tid" is a
 /// case-sensitive string representing a GUID. The presence of this claim and its proper validation
 /// are critical for the security of multi-tenant applications.
-public struct TenantIDClaim: JWTClaim, Equatable {
+public struct TenantIDClaim: JWTClaim, Equatable, ExpressibleByStringLiteral, ExpressibleByNilLiteral {
     /// See ``JWTClaim``.
```
Comment (gwynne) [inline Sources/JWTKit/Claims/TenantIDClaim.swift:11]: When documenting properties or methods which implement protocol requirements, do _not_ use DocC comments; it suppresses autogeneration of the appropriate reference to the protocol.
```suggestion
    // See `JWTClaim.value`.
```

File: Sources/JWTKit/Claims/TenantIDClaim.swift [hunk 3]
```diff
@@ -7,12 +7,20 @@
...
     public var value: String?
 
     /// See ``JWTClaim``.
```
Comment (gwynne) [inline Sources/JWTKit/Claims/TenantIDClaim.swift:14]: When documenting properties or methods which implement protocol requirements, do _not_ use DocC comments; it suppresses autogeneration of the appropriate reference to the protocol.
```suggestion
    // See `JWTClaim.init(value:)`.
```

File: Tests/JWTKitTests/VendorTokenTests.swift
Comment (gwynne) [inline Tests/JWTKitTests/VendorTokenTests.swift:1]: There is no test coverage added for `TenantIDClaim`.

Comment (ptoffy) [reply Tests/JWTKitTests/VendorTokenTests.swift:1]: I'm planning to add test coverage for claims in a separate PR

**Review by gwynne — APPROVED** [review]

**Pull request merged**